src/include/transaction/transaction_context.h

```diff
@@ -155,27 +149,37 @@ class TransactionContext {
    */
   bool IsReadOnly() const { return undo_buffer_.Empty() && loose_ptrs_.empty(); }
 
+  /**
+   * Defers an action to be called if and only if the transaction aborts.  Actions executed LIFO.
+   * @param a the action to be executed. A handle to the system's deferred action manager is supplied
+   * to enable further deferral of actions
+   */
+  void RegisterAbortAction(const TransactionAction &a) { abort_actions_.push_front(a); }
+
   /**
    * Defers an action to be called if and only if the transaction aborts.  Actions executed LIFO.
    * @param a the action to be executed
    */
-  void RegisterAbortAction(const Action &a) { abort_actions_.push_front(a); }
+  void RegisterAbortAction(const std::function<void()> &a) {
+    RegisterAbortAction([=](transaction::DeferredActionManager * /*unused*/) { a(); });
```

Comment:
Do we know how much of a hit we take by wrapping a lambda in a lambda?

Reply:
This part of the code is not performance critical, so I didn't bother benchmarking it. There exists a TODO for this (https://github.com/cmu-db/terrier/blob/ded38bb7629b744eac555c28d7ef731cd06ebdf5/src/include/transaction/deferred_action_manager.h#L42)

Do we have an eager beginner who wants to look at this?

Reply:
Create an issue and mark it beginner.  I’m with you that the penalty is not a major concern for now.